internal/consoleplugin/config.go

```diff
@@ -0,0 +1,223 @@
+)
+
+// Config holds configuration for the logging console plugin.
+// Construct with NewConfig.
+type Config struct {
+	Name      string
+	Namespace string
+	Image     string
+	LokiName  string
+	LokiPort  int32
+	CreatedBy string
+
+	nginxPort, defaultMode, replicas int32
+}
+
+// NewConfig returns a config with default settings.
+// Modify exported fields to change settings.
+func NewConfig(namespace, createdBy string) *Config {
+	return &Config{
+		Name:      "logging-view-plugin",
+		Namespace: namespace,
+		CreatedBy: createdBy,
+		Image:     "quay.io/gbernal/logging-view-plugin:latest", // FIXME
+		LokiName:  "loki",                                       // FIXME
+		LokiPort:  3100,
+
+		nginxPort:   9443,
+		defaultMode: 420,
+		replicas:    1,
+	}
+}
+
+func (cf *Config) secret() string              { return cf.Name + "-secret" }
+func (cf *Config) configMapName() string       { return cf.Name + "-conf" }
+func (cf *Config) selector() map[string]string { return map[string]string{"app": cf.Name} }
+
+const (
+	labelApp             = "app"
+	labelAppK8sName      = "app.kubernetes.io/name"
+	labelAppK8sCreatedBy = "app.kubernetes.io/created-by"
+)
+
+// setup adds common settings for all objects
+func (cf *Config) setup(o client.Object) {
+	l := o.GetLabels()
+	if l == nil {
+		l = map[string]string{}
+	}
+	l[labelApp] = cf.Name
+	l[labelAppK8sName] = cf.Name
+	l[labelAppK8sCreatedBy] = cf.CreatedBy
+	o.SetLabels(l)
+}
+
+func (cf *Config) configMap() *corev1.ConfigMap {
+	o := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cf.configMapName(),
+			Namespace: cf.Namespace,
+		},
+		Data: map[string]string{
+			"nginx.conf": fmt.Sprintf(`
+				error_log /dev/stdout info;
+				events {}
+				http {
+					access_log         /dev/stdout;
+					include            /etc/nginx/mime.types;
+					default_type       application/octet-stream;
+					keepalive_timeout  65;
+					server {
+						listen              %v ssl;
+						ssl_certificate     /var/serving-cert/tls.crt;
+						ssl_certificate_key /var/serving-cert/tls.key;
+						root                /usr/share/nginx/html;
+					}
+				}
+				`, cf.nginxPort)},
+	}
+	cf.setup(o)
+	return o
+}
+
+func (cf *Config) service() *corev1.Service {
+	o := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cf.Name,
+			Namespace: cf.Namespace,
+			Annotations: map[string]string{
+				"service.alpha.openshift.io/serving-cert-secret-name": cf.secret(),
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{{
+				Name:       fmt.Sprintf("%v-tcp", cf.nginxPort),
+				Protocol:   corev1.Protocol("TCP"),
+				Port:       cf.nginxPort,
+				TargetPort: intstr.IntOrString{IntVal: cf.nginxPort},
+			}},
+			Selector:        cf.selector(),
+			Type:            corev1.ServiceType("ClusterIP"),
+			SessionAffinity: corev1.ServiceAffinity("None"),
+		},
+	}
+	cf.setup(o)
+	return o
+}
+
+func (cf *Config) deployment() *appv1.Deployment {
+	o := &appv1.Deployment{
```

Comment:
Same here. We have utilities and builders to do this

Reply:
https://github.com/openshift/cluster-logging-operator/blob/master/internal/k8shandler/deployment.go#L15